java/dagger/internal/codegen/SimpleMethodBindingExpression.java

```diff
@@ -176,13 +176,16 @@ private Expression injectMembers(CodeBlock instance) {
   }
 
   private Optional<CodeBlock> moduleReference(ClassName requestingClass) {
-    return provisionBinding.requiresModuleInstance()
-        ? provisionBinding
-            .contributingModule()
-            .map(Element::asType)
-            .map(ComponentRequirement::forModule)
-            .map(module -> componentRequirementExpressions.getExpression(module, requestingClass))
-        : Optional.empty();
+    if (provisionBinding.requiresModuleInstance(compilerOptions)) {
```

Comment:
FYI that this doesn't change anything about `Factory<>` types. You probably want to adjust those as well.

Reply:
Could you give me an example?

Reply:
https://github.com/google/dagger/blob/2826c31819ae63db6698bd910345ccd0588add10/java/dagger/internal/codegen/InjectionMethods.java#L487-L491 is used to create the proxy method in the factory classes

and also this:
https://github.com/google/dagger/blob/2826c31819ae63db6698bd910345ccd0588add10/java/dagger/internal/codegen/InjectionMethods.java#L501-L508

Btw did you test this locally? https://github.com/bazelbuild/rules_kotlin may be useful